tests/org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/parser/XtextAntlrGeneratorComparisonFragmentTest.xtend

```diff
@@ -0,0 +1,176 @@
+/*******************************************************************************
+ * Copyright (c) 2015 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.generator.parser
+
+import org.eclipse.xtext.generator.parser.antlr.XtextAntlrGeneratorComparisonFragment
+import org.junit.Test
+
+import static org.junit.Assert.*
+
+/**
+ * Contributes unit test for {@link XtextAntlrGeneratorComparisonFragment}.
+ * 
+ * @author Christian Schneider - Initial contribution and API
+ */
+class XtextAntlrGeneratorComparisonFragmentTest extends XtextAntlrGeneratorComparisonFragment {
```

Comment:
That screams for extracting the comparison logic into its own, easily unit-testable class.